### PR TITLE
[master] Merge v8.3 -> v8.4

### PR DESCRIPTION
**Pull request by yaron-zilliqa**
## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ RUN mkdir -p /scilla/0/bin2/ && cp -L /scilla/0/bin/* /scilla/0/bin2/ && strip /
 FROM zilliqa/zilliqa-ccache:${ZILLIQA_CCACHE_VERSION} AS builder
 
 # Manually input tag or commit, can be overwritten by docker build-args
-ARG COMMIT_OR_TAG=master
```
Comment (chetan-zilliqa) [inline docker/Dockerfile:28]: should be `v8.4.0` , right?

File: docker/Dockerfile.deps [hunk 2]
```diff
@@ -24,7 +24,7 @@ ARG VCPKG_COMMIT_OR_TAG=2022.09.27
 ARG VCPKG_ROOT=/vcpkg
 
 # Manually input tag or commit, can be overwritten by docker build-args
-ARG COMMIT_OR_TAG=master
+ARG COMMIT_OR_TAG=v8.3.0
```
Comment (chetan-zilliqa) [inline docker/Dockerfile.deps:27]: same as above.

**Review by chetan-zilliqa — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (yaron-zilliqa) [reply docker/Dockerfile:28]: Yes.. it's usually passed in as a --build-arg though in the command line though.

File: docker/Dockerfile.deps (hunk 2 above)
Comment (yaron-zilliqa) [reply docker/Dockerfile.deps]: Changed.

File: docker/Dockerfile (hunk 1 above)
Comment (JamesHinshelwood) [reply docker/Dockerfile:28]: I don't know if this is correct - `8.4.0` isn't a tag in GitHub. Really I think we should just leave this blank if we're expecting it to be passed as a build-arg?

**Review by chetan-zilliqa — APPROVED** [review]

**Pull request merged**